hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -37,21 +40,44 @@ def validate_class_names_for_plot_options(
     return class_names
 
 
-def save_scores_histogram(results: ResultsType, figures_dir: Path) -> None:
+def save_scores_histogram(results: ResultsType, figures_dir: Path, stage: str = '') -> None:
     """Plots and saves histogram scores figure in its dedicated directory.
 
-    :param results: List that contains slide_level dicts
+    :param results: Dict of lists that contains slide_level results
     :param figures_dir: The path to the directory where to save the histogram scores.
+    :param stage: Test or validation, used to name the figure. Empty string by default.
     """
     fig = plot_scores_hist(results)
-    save_figure(fig=fig, figpath=figures_dir / "hist_scores.png")
+    save_figure(fig=fig, figpath=figures_dir / f"hist_scores_{stage}.png")
 
 
-def save_confusion_matrix(results: ResultsType, class_names: Sequence[str], figures_dir: Path) -> None:
+def save_pr_curve(results: ResultsType, figures_dir: Path, stage: str = '') -> None:
+    """Plots and saves PR curve figure in its dedicated directory. This implementation
+    only works for binary classification.
+''
+    :param results: Dict of lists that contains slide_level results
+    :param figures_dir: The path to the directory where to save the histogram scores
+    :param stage: Test or validation, used to name the figure. Empty string by default.
+    """
+    true_labels = [i.item() if isinstance(i, Tensor) else i for i in results[ResultsKey.TRUE_LABEL]]
+    if len(set(true_labels)) == 2:
+        scores = [i.item() if isinstance(i, Tensor) else i for i in results[ResultsKey.PROB]]
+        fig, ax = plt.subplots()
+        plot_pr_curve(true_labels, scores, legend_label=stage, ax=ax)
+        ax.legend()
+        format_pr_or_roc_axes(plot_type='pr', ax=ax)
+        save_figure(fig=fig, figpath=figures_dir / f"pr_curve_{stage}.png")
+    else:
+        raise Warning("The PR curve plot implementation works only for binary cases, this plot will be skipped.")
```

Comment:
Where is this Warning class coming from? Is it consistent with `logging.warning` that we use elsewhere?

Reply:
Ahh that was related to Valentina's PR, I need to double check it

Reply:
switched it to logging.warning, I don't think we should raise an exception for this, we can just infor the user that it's skipped